### PR TITLE
[IMP] elearning : New Content arrow

### PR DESCRIPTION
**Pull request by laa-odoo**
Have a visual insight that new content has been added on the home page of the courses.

TASK-ID : 2025186

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_slides/views/website_slides_templates_course.xml [hunk 1]
```diff
@@ -427,16 +427,19 @@
             </span>
         </div>
 
-        <div class="d-flex flex-row">
```
Comment (dbeguin) [inline addons/website_slides/views/website_slides_templates_course.xml]: useless diff.
Also, not sure it does change something to move both Free Preview tags.
Or maybe I'm missing something.

File: addons/website_slides/controllers/main.py [hunk 2]
```diff
@@ -240,7 +240,7 @@ def slides_channel_home(self, **post):
         domain = request.website.website_domain()
         channels_all = request.env['slide.channel'].search(domain)
         if not request.env.user._is_public():
-            channels_my = channels_all.filtered(lambda channel: channel.is_member).sorted('completion', reverse=True)[:3]
+            channels_my = channels_all.filtered(lambda channel: channel.is_member).sorted('completion', reverse=True).sorted('completed')[:3]
```
Comment (dbeguin) [inline addons/website_slides/controllers/main.py]: Not sure what you want here :
Among all the channels, you want to have only the 3 first that are completed but with the maximum completion ?
I think if completed is True, completion will be 100% anyway.
So you can just add in you filter the completed flag and take the 3 first.

Note that completed is now a boolean (and not the number of completed slides anymore) --> rebase if necessary
Also, sorting two times in a row is useless, the last sorting only will be taken into account.

File: addons/website_slides/models/slide_channel.py [hunk 3]
```diff
@@ -173,6 +175,20 @@ def _default_access_token(self):
...
+            ('channel_id', 'in', self.ids),
+            ('partner_id', '=', self.env.user.partner_id.id),
+            ('completed', '=', True)])
+        for channel in self:
+            current_is_new_slide_ids = is_new_slide_ids.filtered(lambda slide: slide.channel_id.id == channel.id)
+            current_partner_completed_slides_ids = partner_completed_slide_ids.mapped('slide_id').filtered(lambda slide: slide.channel_id.id == channel.id and not slide.is_category)
+            channel.has_new_content = len(current_is_new_slide_ids) > 0 
+            channel.partner_has_new_content = len(current_is_new_slide_ids - current_partner_completed_slides_ids) > 0
+
+
```
Comment (dbeguin) [inline addons/website_slides/models/slide_channel.py]: remove this line

File: addons/website_slides/views/website_slides_templates_course.xml [hunk 4]
```diff
@@ -428,15 +428,18 @@
         </div>
 
         <div class="d-flex flex-row">
+            <a t-if="channel.can_upload" href="#">
+                <span t-att-data-slide-id="slide.id" t-attf-class="o_wslides_js_slide_toggle_is_preview badge #{'badge-success py-1' if slide.is_preview else 'badge-light badge-hide border'} font-weight-normal px-2 m-1">Free preview</span>
+            </a>
+            <span t-elif="slide.is_preview and not channel.is_member" class="badge badge-success font-weight-normal px-2 py-1 m-1">Free preview</span>
+            <span t-if="slide.is_new_slide and not channel_progress[slide.id].get('completed')" class="badge badge-info badge-arrow-right font-weight-normal px-2 py-1 m-1">
+                New
+            </span>
```
Comment (dbeguin) [inline addons/website_slides/views/website_slides_templates_course.xml]: Are you sure it's necessary to move this block ? I think you can only adapt without moving everything.

Comment (dbeguin) [reply addons/website_slides/views/website_slides_templates_course.xml]: No ok, you can keep it like this, for design bioutifulness reason

File: addons/website_slides/controllers/main.py [hunk 5]
```diff
@@ -240,7 +240,7 @@ def slides_channel_home(self, **post):
         domain = request.website.website_domain()
         channels_all = request.env['slide.channel'].search(domain)
         if not request.env.user._is_public():
-            channels_my = channels_all.filtered(lambda channel: channel.is_member).sorted('completion', reverse=True)[:3]
+            channels_my = channels_all.filtered(lambda channel: channel.is_member).sorted(lambda channel: not channel.completed and -channel.completion)[:3]
```
Comment (dbeguin) [inline addons/website_slides/controllers/main.py]: whut ?  `-channel.completion` Typo ? real shady weird magic stuff ?

Comment (dbeguin) [reply addons/website_slides/controllers/main.py]: `channels_all.filtered(lambda channel: channel.is_member).sorted(lambda channel: 100 if channel.completed else channel.completion, reverse=True)[:3]`

@awa-odoo : Thanks ! ;)

Comment (laa-odoo) [reply addons/website_slides/controllers/main.py]: If a course is completed, we don't want to see it in first position but in last, so I replace 100 by 0

channels_my = channels_all.filtered(lambda channel: channel.is_member).sorted(lambda channel: 0 if channel.completed else channel.completion, reverse=True)[:3]

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**